### DISCUSSION TITLE
How to override a submodule?

### DISCUSSION THREAD

**Discussion by juhp** [magit/magit#5152] [Ideas]
Sometimes as a hack I change a submodule working directory to a symlink to a checkout of its repo.
However today I realised then magit-status silently stops showing the local changes because the submodule is now in a "bad state".

I think running `git status` by hand gave me `error: expected submodule path 'the-submodule-in-question' not to be a symbolic link`.

Before opening a bug report I thought I should ask here first what is the correct way to do this?

(For a moment I was worried all my local changes had suddenly been lost and forgotten!?!)

**Comment by tarsius** [comment]
> Before opening a bug

I assume the bug report would be "there's this state, which git cannot handle, and magit silently ignores it, instead of raising the error to the user".

I would classify this as a minor bug (or even an enhancement), and at present time these get a "I am trying to get releases out the door. I won't be working on this until some time after that is done. And it could be a while, because there are other more important things already in the backlog." response.

Instead I've done so here, and added a bookmark in my "follow up" folder.

> How to override a submodule?

As far as I understand you do: `rm path/to/module; ln -s /path/to/fullrepo path/to`. If so, don't; git doesn't support that. Instead to `cd path/to/module; git checkout what-is-needed-temporarily`.

**Reply by juhp** [reply]
Thank you - totally makes sense
